src/Avalonia.Controls/Control.cs

```diff
@@ -329,5 +347,55 @@ protected override void OnKeyUp(KeyEventArgs e)
                 }
             }
         }
+
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == FlowDirectionProperty)
+            {
+                InvalidateFlowDirection();
+                
+                foreach (var visual in VisualChildren)
+                {
+                    if (visual is Control child)
+                    {
+                        child.InvalidateFlowDirection();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Computes the <see cref="IVisual.IsMirrorTransform"/> value according to the 
+        /// <see cref="FlowDirection"/> and <see cref="BypassFlowDirectionPolicies"/>
+        /// </summary>
+        public virtual void InvalidateFlowDirection()
+        {
+            FlowDirection thisFD = this.FlowDirection;
```

Comment:
Please use `var` here. Also please change the naming `flowDirection` `parentFlowDirection`

Reply:
I changed to `var,` but why parent?

Reply:
```suggestion
            FlowDirection flowDirection = this.FlowDirection;
```

Reply:
ok